src/main/java/opendcs/dao/LoadingAppDao.java

```diff
@@ -531,69 +532,78 @@ public TsdbCompLock obtainCompProcLock(CompAppInfo appInfo, int pid, String host
 	{
 		//ResultSet rs = null; //doQuery(q);
 		TsdbCompLock lock = null;
-		try( PreparedStatement getLock =
-				getConnection().prepareStatement(
-					"SELECT * from CP_COMP_PROC_LOCK WHERE LOADING_APPLICATION_ID = ?"
-				);
-		)
-		{
-			getLock.setLong(1,appInfo.getAppId().getValue());
-			try(ResultSet rs = getLock.executeQuery();){
-				if ( rs.next() && (lock = rs2lock(rs)) != null)
+		try (Connection c = getConnection();)
```

Comment:
Is the nested try necessary?  can both connection and getLock be created in same try?

Reply:
No, because you can't do the setLong in the try(...) block. Only assignments to an object implementing AutoClosable.

At least that's been my experience. This is actually one of the reasons I created the new handles and tools like JDBI exist... the nesting gets really annoying sometimes.